alf/algorithms/mcts_algorithm.py

```diff
@@ -194,6 +200,27 @@ class MCTSAlgorithm(OffPolicyAlgorithm):
     2. `Hubert et. al. Learning and Planning in Complex Action Spaces
     <https://arxiv.org/abs/2104.06303>`_
       It is enabled when SimpleMCTSModel.num_sampled_actions is set.
+
+    The time spent on tree search is directly related to how many times the tree
+    is expanded. To make it faster, we also support expanding multiple
+    leaves simulaneously. In order to do this, we maintain num_parallel_sims best
+    children for each node in the tree and use them to contruct k=num_paralilel_sims
+    paths. Note the k best children may have duplicates, which is desired because
+    we want to expand the most promising path more often. Depending the value of
+    ``search_with_exploration_policy``, this process is slightly different:
+
+    * search_with_exploration_policy=True. The k best_children of each node are
+      simply chosen by independently sampling the exploration policy k times.
+      When contructing the search paths, the i-th search path is based on the
+      i-th best child of each node.
+
+    * search_with_exploration_policy=False. The best child is same the case k=1.
+      The second best child is found by assuming the visit count of the best
+      child and the parent are increased by 1 and applying the UCB criterion
+      again. This is repeated k times to get k best children. During the process
+      of traversing from the root to contruct k search paths, if the current end
+      node of several (let's say k') paths are the same, we will use best k'
+      children of that node to extend the paths.
```

Comment:
When I reading this paragraph without reading the code yet, I had a few questions:

1. Is this procedure seems "not parallel" across `k`?
2. "second best" is a bit ambiguous here, it can mean "the one that is only worse to the best one" or "the best that is picked in the second sampling".
3. Had some hard time understanding this "if the current end node of several (let's say k') paths are the same, we will use best k' children of that node to extend the paths". Some Ascii graph might help. Also I am confused whether this behavior is
   * a special treatment or,
   * is implied from the procedure but highlighted